pkg/cmd/roachtest/main.go

```diff
@@ -515,4 +513,14 @@ func validateAndConfigure(cmd *cobra.Command, args []string) {
 				roachtestflags.UseSpotVM, spotFlagInfo.Name, spotFlagInfo.Usage))
 		}
 	}
+
+	// Test selection and select probability flags are mutually exclusive. If both are set,
+	// override selective-tests to false and only use select-probability. It's likely that
+	// we're running on a TC branch that defaults selective-tests to true, and a user manually
+	// overrode select-probability and wishes to use that instead.
+	selectProbFlagInfo := roachtestflags.Changed(&roachtestflags.SelectProbability)
+	if roachtestflags.SelectiveTests && selectProbFlagInfo != nil {
+		fmt.Fprintf(os.Stderr, "WARN: select-probability and selective-tests=true are incompatible. Disabling selective-tests.\n")
```

Comment:
Alternatively, we could just fail out, but my concern with that if on TC, you've now just wasted 40 minutes of build time just to find out the flags aren't compatible.

Reply:
We could duplicate this check in the TC (wrapper) script and fail the build before it has a chance to build.

Reply:
Tried it out and it works fine, but I realized it might be unobvious where/why the `select-probability` and `selective-tests` are being set since they are most likely set by the TC script and not the user.

Leaning towards merging as is unless anyone has any strong opinions.

Reply:
> I realized it might be unobvious where/why the select-probability and selective-tests are being set since they are most likely set by the TC script and not the user.

Who is the user? (TC is technically one.) If the unobvious part is an override which happens due to a specific release branch, we can `echo` it out explicitly. So, the human (user) can see the build log and tell immediately why we bailed out.

Reply:
I feel like if the user doesn't have a choice anyways, it makes more sense to auto-fix and display a warning rather than erroring out. This would just make the TC workers frustrated!

Reply:
Yeah Ludo's thinking was the same as mine, but after some more thought I think it's bit of a contrived concern. Engineers shouldn't be manually kicking off nightlies on master/release branches anyway (since failures will create github issues). They will only run into "you've now just wasted 40 minutes of build time just to find out the flags aren't compatible" if they are doing something unsupported.

I feel like we can just change the referenced line to an error out, without the need for an early check in the script. Thoughts?

Reply:
> They will only run into "you've now just wasted 40 minutes of build time just to find out the flags aren't compatible" if they are doing something unsupported.

Yes, I like it! It's essentially punishment for doing the wrong thing in the first place. :)